src/stores/room-list-v3/skip-list/RoomNode.ts

```diff
@@ -48,4 +49,31 @@ export class RoomNode {
         const activeSpace = SpaceStore.instance.activeSpace;
         this._isInActiveSpace = SpaceStore.instance.isRoomInSpace(activeSpace, this.room.roomId);
     }
+
+    /**
+     * Aggregates all the filter keys that apply to this room.
+     * eg: if filterKeysMap.get(Filter.FavouriteFilter) is true, then this room is a favourite room.
+     */
+    private filterKeysMap: Map<FilterKey, boolean> = new Map();
```

Comment:
why not use a `Set`? has the same semantics & complexity but uses less space in the minimum case

Reply:
https://github.com/element-hq/element-web/pull/29433/commits/2ab117dd3e23700c32116fa874dff5a1b0d8f440